client/gutenberg/editor/index.js

```diff
@@ -1,11 +1,28 @@
 import page from 'page';
-import { makeLayout, render as clientRender } from 'calypso/controller';
+import { makeLayout, redirectLoggedOut, render as clientRender } from 'calypso/controller';
+import { getSiteFragment } from 'calypso/lib/route';
 import { siteSelection, sites } from 'calypso/my-sites/controller';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getImmediateLoginLocale } from 'calypso/state/immediate-login/selectors';
 import { authenticate, post, redirect, siteEditor, exitPost } from './controller';
 
+function redirectToPermalinkIfLoggedOut( context, next ) {
```

Comment:
Do you mind moving this one to `client/gutenberg/editor/controller.js`? That's where the rest of the similar controller functions live.

Reply:
Good call.

Reply:
Done